From c3c17a3cd6b1b9efd3c15b5c686747502732de50 Mon Sep 17 00:00:00 2001
From: Thanneeru Srinivasulu <tsrinivasulu@caviumnetworks.com>
Date: Wed, 15 Jul 2015 12:44:22 +0530
Subject: [PATCH 34/59] net: mdio-octeon: thunderx: Force to load octeon-mdio
 before bgx driver.
Patch-mainline: v4.4-rc4
Git-commit: 723cda5b055851f5e8bf61aacd8008c43c99e801

PHYs need to be ready before bgx probing.

Signed-off-by: Thanneeru Srinivasulu <tsrinivasulu@caviumnetworks.com>
Signed-off-by: Robert Richter <rrichter@cavium.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/net/ethernet/cavium/thunder/thunder_bgx.c |    3 +++
 drivers/net/ethernet/cavium/thunder/thunder_bgx.h |    1 +
 2 files changed, 4 insertions(+)

--- a/drivers/net/ethernet/cavium/thunder/thunder_bgx.c
+++ b/drivers/net/ethernet/cavium/thunder/thunder_bgx.c
@@ -1027,6 +1027,9 @@ static int bgx_probe(struct pci_dev *pde
 	struct bgx *bgx = NULL;
 	u8 lmac;
 
+	/*Load octeon mdio driver*/
+	octeon_mdiobus_force_mod_depencency();
+
 	bgx = devm_kzalloc(dev, sizeof(*bgx), GFP_KERNEL);
 	if (!bgx)
 		return -ENOMEM;
--- a/drivers/net/ethernet/cavium/thunder/thunder_bgx.h
+++ b/drivers/net/ethernet/cavium/thunder/thunder_bgx.h
@@ -182,6 +182,7 @@ enum MCAST_MODE {
 #define BCAST_ACCEPT	1
 #define CAM_ACCEPT	1
 
+void octeon_mdiobus_force_mod_depencency(void);
 void bgx_lmac_rx_tx_enable(int node, int bgx_idx, int lmacid, bool enable);
 void bgx_add_dmac_addr(u64 dmac, int node, int bgx_idx, int lmac);
 unsigned bgx_get_map(int node);
